handlers/metrics.go

```diff
@@ -228,34 +248,32 @@ func ControlPlaneMetrics(promSupplier promClientSupplier) http.HandlerFunc {
 	}
 }
 
-// ResourceUsageMetrics is the API handler to fetch metrics to be displayed, related to a single control plane revision
-func ResourceUsageMetrics(promSupplier promClientSupplier) http.HandlerFunc {
+// ZtunnelMetrics is the API handler to fetch metrics to be displayed, related to a single control plane revision
+func ResourceUsageMetrics(conf *config.Config, cache cache.KialiCache, discovery *istio.Discovery, clientFactory kubernetes.ClientFactory, prom prometheus.ClientInterface) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		vars := mux.Vars(r)
 		namespace := vars["namespace"]
 		app := vars["app"]
-		cluster := clusterNameFromQuery(config.Get(), r.URL.Query())
+		cluster := clusterNameFromQuery(conf, r.URL.Query())
 
-		metricsService, namespaceInfo := createMetricsServiceForNamespaceMC(w, r, promSupplier, namespace)
-		if metricsService == nil {
+		namespaceInfo, err := checkNamespaceAccess(w, r, conf, cache, discovery, clientFactory, namespace, cluster)
+		if err != nil {
 			// any returned value nil means error & response already written
 			return
 		}
-		oldestNs := GetOldestNamespace(namespaceInfo)
-
-		params := models.IstioMetricsQuery{App: app, Cluster: cluster, Namespace: namespace}
 
-		err := extractIstioMetricsQueryParams(r, &params, oldestNs)
-		if err != nil {
+		params := models.IstioMetricsQuery{App: app, Cluster: cluster, Namespace: namespaceInfo.Name}
+		if err := extractIstioMetricsQueryParams(r, &params, namespaceInfo); err != nil {
 			RespondWithError(w, http.StatusBadRequest, err.Error())
 			return
 		}
 
-		if namespace != config.Get().IstioNamespace {
+		if namespace != conf.IstioNamespace {
 			RespondWithError(w, http.StatusBadRequest, fmt.Sprintf("namespace [%s] is not the control plane namespace", namespace))
 			return
 		}
 
+		metricsService := business.NewMetricsService(prom, conf)
 		metrics := make(models.MetricsMap)
 
 		ztunnelMetrics, err := metricsService.GetResourceMetrics(params)
```

Comment:
It would be more accurate to rename ztunnelMetrics here - It looks that was missing from refactor